### PR TITLE
Make structural copy handle Blists properly.

### PR DESCRIPTION
**Pull request by stevelinton**
Addresses #1428

### CODE REVIEW

File: src/blister.c [hunk 1]
```diff
@@ -284,17 +284,31 @@ Obj DoCopyBlist(Obj list, Int mut) {
...
-    /* don't change immutable objects                                      */
-    if ( ! IS_MUTABLE_OBJ(list) ) {
-        return list;
-    }
-
-    return DoCopyBlist(list, mut);
+  Obj copy;
+  Obj tmp;
+ 
+    copy =  DoCopyBlist(list, mut);
```
Comment (fingolfin) [inline src/blister.c]: Why is the declaration of the locals indented differently than the rest of the code? (For that matter: The other function you introduce indent by 2 spaces, this function by 4, mostly)

File: src/blister.c [hunk 2]
```diff
@@ -334,7 +348,7 @@ void CleanBlistCopy (
     Obj                 list )
 {
     /* remove the forwarding pointer                                       */
-    ADDR_OBJ(list)[0] = ADDR_OBJ( ADDR_OBJ(list)[0] )[0];
+    ADDR_OBJ(list)[0] = ADDR_OBJ( ADDR_OBJ(list)[0] )[1];
```
Comment (fingolfin) [inline src/blister.c]: Shouldn't this be `ADDR_OBJ(list)[0] = ELM_PLIST( ADDR_OBJ(list)[0], 1 );` ?

File: src/blister.c [hunk 3]
```diff
@@ -312,7 +326,7 @@ Obj CopyBlistCopy (
     Obj                 list,
     Int                 mut )
 {
-    return ADDR_OBJ(list)[0];
+  return ADDR_OBJ(ADDR_OBJ(list)[0])[2];
```
Comment (fingolfin) [inline src/blister.c]: Shouldn't this be `return ELM_PLIST( ADDR_OBJ(list)[0], 2 );` ?

File: src/blister.c (hunk 1 above)
Comment (stevelinton) [inline src/blister.c]: Insufficiently clever emacs settings. Will fix

File: src/blister.c (hunk 3 above)
Comment (stevelinton) [inline src/blister.c]: Yes

File: src/blister.c (hunk 2 above)
Comment (stevelinton) [inline src/blister.c]: Yes

**Comment by fingolfin** [comment]
@stevelinton do you plan on updating this PR? Or should somebody else take it over?

**Comment by markuspf** [comment]
This has been superseded by #1514.

**Pull request closed without merging**